app/controllers/users_controller.rb

```diff
@@ -0,0 +1,16 @@
+class UsersController < ApplicationController
+  before_action :authenticate_user!, except: [:show]
+
+  def withdrawal
+    if current_user.update(is_deleted: true)
+      reset_session
+      flash[:notice] = "退会処理を実行いたしました"
+      redirect_to root_path
+    else
+      render user_path
```

Comment:
renderにはpath helperではなくアクション名を指定するはずです。

```suggestion
      render :show
```